typescript/introduction/intro-typescript/why-typescript.md

```diff
@@ -0,0 +1,53 @@
+---
+author: kapnobatai136
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+
+---
+
+# Why TypeScript?
+
+---
+## Content
+
+The main feature, as its name suggests, is that it allows the use of *static types*. 
+
+The static in this context means that types are checked by looking at the written code *without needing to run it*. 
+
+In other words, TypeScript checks your code by making sure you use the same names and shapes where you said you would.
+
+You can also think of it as a **spellchecker for JavaScript**.
+
+This is useful because making sure that the same types of data are used throughout the code can help avoid bugs. It makes the code consistent, and less prone to typos.
+
+It also makes the development faster because you can catch problems early on. 
+
+Since the code doesn't need to run to be type-checked (remember, static types), your editor can integrate with TypeScript and underline in red any type mismatches in the code as you're writing it. This means you write safer code from the start.
+
+![ts-safer-code](https://img.enkipro.com/1dca4c660ac1affdf9e903db249599ef.png)
+Since TypeScript is introduced as another layer on top of JavaScript, it also allows us to use the latest JavaScript features and handles the conversion for old platforms behind the scenes.
```

Comment:
Yeah, TS team is pretty strict about rejecting features not accepted by ES community.
Look how the massacred my boy https://github.com/microsoft/TypeScript/issues/9

Reply:
What made you think if this? Do you maybe have a suggestion in mind how we can add it without introducing a lot more explanation?

Reply:
optional chaining is supported by TS 3.7 & is in ES stage 4, but it's no in any official releases of javascript https://www.w3schools.com/js/js_versions.asp

Reply:
Optional chaining is in ES2020. (Note that W3 schools is not a credible source)

Reply:
Anything in Stage 4 by default is approved to be released as an ESNext version

Reply:
I do agree with both of your positions though, we should mention that TS is ES-compliant because it aims to maintain full inter-operability with JS (which is ES-compliant)

Reply:
If only ES could move faster. I'm wanting for extension functions for 3 years now.